jujugui/static/gui/src/app/store/env/acl.js

```diff
@@ -43,48 +43,55 @@ YUI.add('acl', function(Y) {
     @param modelAPI {Object} The model API connection instance.
     @return {Object} A namespace providing access to ACLs checkers (see below).
   */
-  juju.generateAcl = (controllerAPI, modelAPI) => ({
-
-    /**
-      Report whether the model interaction is read-only, in which case it is
-      not possible to interact with the model, just watch it.
-
-      @function isReadOnly
-      @return {Boolean} Whether the user has read-only access.
-    */
-    isReadOnly: () => modelAPI.get('modelAccess') === 'read',
-
-    /**
-      Report whether the current user can create models.
-
-      @function canAddModels
-      @return {Boolean} Whether the user can add models.
-    */
-    canAddModels: () => {
-      const access = (
-        modelAPI.get('controllerAccess') ||
-        controllerAPI.get('controllerAccess')
-      );
-      return access === 'addmodel' || access === 'superuser';
-    },
-
-    /**
-      Report whether the user can share the current model.
-
-      @function canShareModel
-      @return {Boolean} Whether the user can share the current model.
-    */
-    canShareModel: () => modelAPI.get('modelAccess') === 'admin',
-
-    /**
-      Report whether the user can destroy the current model.
-
-      @function canRemoveModel
-      @return {Boolean} Whether the user can destroy the current model.
-    */
-    canRemoveModel: () => modelAPI.get('modelAccess') === 'admin'
-
-  });
+  juju.generateAcl = (controllerAPI, modelAPI) => {
+    if (!controllerAPI) {
+      // The application is connected to a legacy Juju.
+      controllerAPI = {
+        get: (_) => ''
```

Comment:
I liked to be clear about the number of expected arguments. I know this is a JS feature, but it feels like a JS bug ;-)